google-cloud-spanner/src/test/java/com/google/cloud/spanner/BenchmarkingDataLoaderUtility.java

```diff
@@ -0,0 +1,173 @@
+   * the instance/database/table is already created. It does not perform any admin operations.
+   *
+   * <p>Table schema used here: CREATE TABLE FOO ( id INT64 NOT NULL, BAZ INT64, BAR INT64, )
+   * PRIMARY KEY(id);
+   */
+  @Test
+  public void bulkInsertTestData() {
+    String insertQuery = "INSERT INTO FOO(id, BAZ, BAR) VALUES(@key, @val1, @val2)";
+    int key = 0;
+    for (int batch = 0; batch < 1000; batch++) {
+      List<Statement> stmts = new LinkedList<>();
+      for (int i = 0; i < 100; i++) {
+        stmts.add(
+            Statement.newBuilder(insertQuery)
+                .bind("key")
+                .to(key)
+                .bind("val1")
+                .to(1)
+                .bind("val2")
+                .to(2)
+                .build());
+        key++;
+      }
+      final TransactionCallable<long[]> callable = transaction -> transaction.batchUpdate(stmts);
+      TransactionRunner runner = client.readWriteTransaction();
+      long[] actualRowCounts = runner.run(callable);
+      assertThat(actualRowCounts.length).isEqualTo(100);
+      System.out.println("Completed batch => " + batch);
+    }
+  }
+
+  /**
+   * Collects all results from a collection of future objects.
+   *
+   * @param service
+   * @param results
+   * @param numOperations
+   * @return
+   * @throws Exception
+   */
+  public static List<Duration> collectResults(
+      final ListeningScheduledExecutorService service,
+      final List<ListenableFuture<List<Duration>>> results,
+      final int numOperations)
+      throws Exception {
+    service.shutdown();
+    if (!service.awaitTermination(60L, TimeUnit.MINUTES)) {
+      throw new TimeoutException();
+    }
+    List<Duration> allResults = new ArrayList<>(numOperations);
+    for (Future<List<Duration>> result : results) {
+      allResults.addAll(result.get());
+    }
+    return allResults;
+  }
+
+  /**
+   * Utility to print latency numbers. It computes metrics such as Average, P50, P95 and P99.
+   *
+   * @param results
+   */
+  public static void printResults(List<Duration> results) {
+    if (results == null) {
+      return;
+    }
+    List<Duration> orderedResults = new ArrayList<>(results);
+    Collections.sort(orderedResults);
+    System.out.println();
+    System.out.printf("Total number of queries: %d\n", orderedResults.size());
+    System.out.printf("Avg: %fs\n", avg(results));
+    System.out.printf("P50: %fs\n", percentile(50, orderedResults));
+    System.out.printf("P95: %fs\n", percentile(95, orderedResults));
+    System.out.printf("P99: %fs\n", percentile(99, orderedResults));
+  }
+
+  public static double percentile(int percentile, List<Duration> orderedResults) {
+    int index = percentile * orderedResults.size() / 100;
+    Duration value = orderedResults.get(index);
+    Double convertedValue = convertDurationToFractionInSeconds(value);
+    return convertedValue;
+  }
+
+  public static double avg(List<Duration> results) {
```

Comment:
Add a Javadoc explaining that the returned value is the average in seconds (the latter is the part that is not self-explanatory)

Reply:
Done